scipy/stats/_continuous_distns.py

```diff
@@ -4076,6 +4076,101 @@ def fit(self, data, *args, **kwds):
 laplace = laplace_gen(name='laplace')
 
 
+class laplace_asymmetric_gen(rv_continuous):
+    r"""An asymmetric Laplace continuous random variable.
+
+    %(before_notes)s
+
+    Notes
+    -----
+    The probability density function for `laplace_asymmetric` is
+
+    .. math::
+
+        f(x, \kappa) &= \frac{1}{\kappa+\kappa^{-1}}\exp(-x\kappa),\quad x\ge0\\
+                     &= \frac{1}{\kappa+\kappa^{-1}}\exp(x/\kappa),\quad x<0\\
+
+    for :math:`-\infty < x < \infty`, :math:`\kappa > 0`.
+
+    `laplace_asymmetric` takes ``kappa`` as a shape parameter for
+    :math:`\kappa`. For :math:`\kappa = 1`, it is identical to a
+    Laplace distribution.
+
+    %(after_notes)s
+
+    See Also
+    --------
+    laplace : Laplace distribution
+
+    References
+    ----------
+    .. [1] Kozubowski TJ and Podg\'orski K. A Multivariate and
+           Asymmetric Generalization of Laplace Distribution,
+           Computational Statistics 15, 531--540 (2000).
+           :doi:`10.1007/PL00022717`
+
+    %(example)s
+
+    """
+    def _rvs(self, kappa, size=None, random_state=None):
+        kapinv = 1/kappa
+        u = -kappa + (kappa+kapinv)*random_state.uniform(size=size)
+        return -np.where(u >= 0,
+            kapinv*np.log1p(-u*kappa),
+            -kappa*np.log1p(u*kapinv))
+
+    def _pdf(self, x, kappa):
+        return np.exp(self._logpdf(x, kappa))
+
+    def _logpdf(self, x, kappa):
+        kapinv = 1/kappa
+        lPx = np.where(x >= 0, -x*kappa, x*kapinv)
```

Comment:
Re: `_rvs`, does this simply mean deleting this redefined method from the class? How  then is  the shape parameter handled (out of curiosity?)

Reply:
Yes. If you remove your `_rvs` override, then the default implementation will be used. Here's the default implementation (top comments removed):
```
    def _rvs(self, *args, size=None, random_state=None):
        ## Use basic inverse cdf algorithm for RV generation as default.
        U = random_state.uniform(size=size)
        Y = self._ppf(U, *args)
        return Y
```
It handles the shape by passing it into `_ppf`. I think the results will be almost identical because this is basically what your `_rvs` method already does (except the ppf code is rewritten a bit differently).